From 880c5ce4c2d2780cabd2ab9f803553ffa7e0bb6b Mon Sep 17 00:00:00 2001
From: "Arumugam,Kamenee" <kamenee.arumugam@intel.com>
Date: Fri, 27 Feb 2015 14:38:24 +0800
Subject: [PATCH 04/32] TBD: serial: serial console hack

This is temporary hack to make Quark serial console works.
It is inherited from Bryan O'Donoghue.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
Signed-off-by: Arumugam,Kamenee <kamenee.arumugam@intel.com>
Signed-off-by: Pengyu Ma <pengyu.ma@windriver.com>
---
 arch/x86/include/asm/serial.h |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/x86/include/asm/serial.h b/arch/x86/include/asm/serial.h
index 9bcaf85..815a730 100644
--- a/arch/x86/include/asm/serial.h
+++ b/arch/x86/include/asm/serial.h
@@ -10,8 +10,11 @@
  * clock, since the 16550A is capable of handling a top speed of 1.5
  * megabits/second; but this requires the faster clock.
  */
-#ifndef BASE_BAUD
+
+#if 0
 #define BASE_BAUD ( 1843200 / 16 )
+#else
+#define BASE_BAUD 2764800
 #endif
 
 /* Standard COM flags (except for COM4, because of the 8514 problem) */
@@ -23,13 +26,16 @@
 #define STD_COM4_FLAGS ASYNC_BOOT_AUTOCONF
 #endif
 
-#ifndef SERIAL_PORT_DFNS
+#if 0
 #define SERIAL_PORT_DFNS			\
 	/* UART CLK   PORT IRQ     FLAGS        */			\
 	{ 0, BASE_BAUD, 0x3F8, 4, STD_COM_FLAGS },	/* ttyS0 */	\
 	{ 0, BASE_BAUD, 0x2F8, 3, STD_COM_FLAGS },	/* ttyS1 */	\
 	{ 0, BASE_BAUD, 0x3E8, 4, STD_COM_FLAGS },	/* ttyS2 */	\
 	{ 0, BASE_BAUD, 0x2E8, 3, STD_COM4_FLAGS },	/* ttyS3 */
+
+#else
+#define SERIAL_PORT_DFNS
 #endif
 
 #endif /* _ASM_X86_SERIAL_H */
-- 
1.7.5.4

